packages/token-bridge-sdk/src/util/withdrawals.ts

```diff
@@ -0,0 +1,229 @@
+    event,
+    l2ChainId: l2ChainID
+  })
+
+  const executedMessagesCache = JSON.parse(
+    localStorage.getItem('arbitrum:bridge:executed-messages') || '{}'
+  )
+  if (executedMessagesCache[cacheKey]) {
+    return OutgoingMessageState.EXECUTED
+  }
+
+  const messageReader = new L2ToL1MessageReader(l1Provider, event)
+
+  try {
+    return await messageReader.status(l2Provider)
+  } catch (error) {
+    return OutgoingMessageState.UNCONFIRMED
+  }
+}
+
+export function getExecutedMessagesCacheKey({
+  event,
+  l2ChainId
+}: {
+  event: L2ToL1EventResult
+  l2ChainId: number
+}) {
+  return isClassicL2ToL1TransactionEvent(event)
+    ? `l2ChainId: ${l2ChainId}, batchNumber: ${event.batchNumber.toString()}, indexInBatch: ${event.indexInBatch.toString()}`
+    : `l2ChainId: ${l2ChainId}, position: ${event.position.toString()}`
+}
+
+export async function attachNodeBlockDeadlineToEvent(
+  event: L2ToL1EventResultPlus,
+  l1Provider: Provider,
+  l2Provider: Provider
+) {
+  if (
+    event.outgoingMessageState === OutgoingMessageState.EXECUTED ||
+    event.outgoingMessageState === OutgoingMessageState.CONFIRMED
+  ) {
+    return event
+  }
+
+  const messageReader = L2ToL1MessageReader.fromEvent(l1Provider, event)
+
+  try {
+    const firstExecutableBlock = await messageReader.getFirstExecutableBlock(
+      l2Provider
+    )
+
+    return { ...event, nodeBlockDeadline: firstExecutableBlock?.toNumber() }
+  } catch (e) {
+    const expectedError = "batch doesn't exist"
+    const expectedError2 = 'CALL_EXCEPTION'
+    const err = e as Error & { error: Error }
+    const actualError = err && (err.message || (err.error && err.error.message))
+    if (actualError.includes(expectedError)) {
+      const nodeBlockDeadline: NodeBlockDeadlineStatus = 'NODE_NOT_CREATED'
+      return {
+        ...event,
+        nodeBlockDeadline
+      }
+    } else if (actualError.includes(expectedError2)) {
+      // in classic we simulate `executeTransaction` in `hasExecuted`
+      // which might revert if the L2 to L1 call fail
+      const nodeBlockDeadline: NodeBlockDeadlineStatus =
+        'EXECUTE_CALL_EXCEPTION'
+      return {
+        ...event,
+        nodeBlockDeadline
+      }
+    } else {
+      throw e
+    }
+  }
+}
+
+export async function tryFetchLatestSubgraphBlockNumber(
+  l2ChainID: number
+): Promise<number> {
+  try {
+    return await fetchL2BlockNumberFromSubgraph(l2ChainID)
+  } catch (error) {
+    // In case the subgraph is not supported or down, fall back to fetching everything through event logs
+    return 0
+  }
+}
+
+export async function mapTokenWithdrawalFromEventLogsToL2ToL1EventResult(
+  result: WithdrawalInitiated,
+  l1Provider: Provider,
+  l2Provider: Provider,
+  l2ChainID: number,
+  walletAddress: string
+): Promise<L2ToL1EventResultPlus | undefined> {
+  const { symbol, decimals } = await getL1TokenData({
+    account: walletAddress,
+    erc20L1Address: result.l1Token!,
```

Comment:
I would add a check here rather than `!`

Reply:
Not needed now. This will always be present after updated ts mappings from sdk